HW3/P3/sum.cl

```diff
@@ -24,8 +29,17 @@ __kernel void sum_coalesced(__global float* x,
     // You can assume get_local_size(0) is a power of 2.
     //
     // See http://www.nehalemlabs.net/prototype/blog/2014/06/16/parallel-programming-with-opencl-and-python-parallel-reduce/
-    for (;;) { // YOUR CODE HERE
-        ; // YOUR CODE HERE
+    loglSize = 1;
+    temp  = lSize >> 1;
+    while (temp > 1){
+        temp = temp >> 1;
+        loglSize = loglSize + 1;
+    }
```

Comment:
Instead of using the above while loop, you can incorporate this to following for loop,
by setting the initial j to lSize >> 1 or (lSize/2), and decrease it by half on each iteration.